plugins/query-insights/src/main/java/org/opensearch/plugin/insights/rules/model/Attribute.java

```diff
@@ -75,6 +80,69 @@ static void writeTo(final StreamOutput out, final Attribute attribute) throws IO
         out.writeString(attribute.toString());
     }
 
+    /**
+     * Write Attribute value to a StreamOutput
+     * @param out the StreamOutput to write
+     * @param attributeValue the Attribute value to write
+     */
+    public static void writeValueTo(StreamOutput out, Object attributeValue) throws IOException {
+        if (attributeValue instanceof SearchSourceBuilder) {
+            ((SearchSourceBuilder) attributeValue).writeTo(out);
+        } else {
+            out.writeGenericValue(attributeValue);
+        }
+    }
+
+    /**
+     * Read attribute value from the input stream given the Attribute type
+     *
+     * @param in the streaminput to read
+     * @param attribute attribute type to differentiate between Source and others
+     * @return parse value
+     * @throws IOException IOException
+     */
+    public static Object readAttributeValue(StreamInput in, Attribute attribute) throws IOException {
+        if (attribute == Attribute.SOURCE) {
+            SearchSourceBuilder builder = new SearchSourceBuilder(in);
+            return builder;
+        } else {
+            return in.readGenericValue();
+        }
+    }
+
+    /**
+     * Read attribute map from the input stream
+     *
+     * @param in the streaminput to read
+     * @return parsed attribute map
+     * @throws IOException IOException
+     */
+    public static Map<Attribute, Object> readAttributeMap(StreamInput in) throws IOException {
+        int size = readArraySize(in);
+        if (size == 0) {
+            return Collections.emptyMap();
+        }
+        Map<Attribute, Object> map = new HashMap<>(size);
+
+        for (int i = 0; i < size; i++) {
+            Attribute key = readFromStream(in);
+            Object value = readAttributeValue(in, key);
+            map.put(key, value);
+        }
+        return map;
+    }
+
+    private static int readArraySize(StreamInput in) throws IOException {
```

Comment:
same as https://github.com/opensearch-project/OpenSearch/pull/14528/files#r1654227717

Reply:
I added them here because you already had the following similar functions in attribute and wanted to keep them all in the same place:
```
    /**
     * Read an Attribute from a StreamInput
     *
     * @param in the StreamInput to read from
     * @return Attribute
     * @throws IOException IOException
     */
    static Attribute readFromStream(final StreamInput in) throws IOException {
        return Attribute.valueOf(in.readString().toUpperCase(Locale.ROOT));
    }

    /**
     * Write Attribute to a StreamOutput
     *
     * @param out the StreamOutput to write
     * @param attribute the Attribute to write
     * @throws IOException IOException
     */
    static void writeTo(final StreamOutput out, final Attribute attribute) throws IOException {
        out.writeString(attribute.toString());
    }
```

Reply:
The above functions are for serializing and deserializing an `Attribute` Enum, for `Attribute` Enum it shouldn't care about the `Value` concept since the `Value` concept exists in `SearchQueryRecord` and should be decoupled from  `Attribute`. 

As mentioned above could you give some examples if these are actually needed, if so we should at least move them to the record class.